local-cli/server/runServer.js

```diff
@@ -157,7 +157,7 @@ function getPackagerServer(args, config) {
     maxWorkers: args.maxWorkers,
     platforms: defaultPlatforms.concat(args.platforms),
     polyfillModuleNames: config.getPolyfillModuleNames(),
-    polyfills: config.polyfills,
+    getPolyfills: config.getPolyfills,
```

Comment:
I'd suggest performing the call at this point; i.e. `polyfills: config.getPolyfills()`. The reason is that the `metro-bundler` machinery is already ready to accept a list of custom polyfills, but as an `Array<string>`, not as a `() => Array<string>`.